### PR TITLE
Fix Preference Loss and Refactor for Readability 

### PR DESCRIPTION
**Pull request by austin362667**
## Summary

Thanks to @winglian and @shivam15s noticed and fixed this https://github.com/linkedin/Liger-Kernel/pull/481.

This PR suggests negating the preference loss terms to align with the formulas in the docstrings, while maintaining the base preference structure as `nll_loss + preference_loss`. This would make our loss computations more consistent since both terms would represent losses to be minimized.

[UPDATE: It seems like being addressed now in [here](https://github.com/linkedin/Liger-Kernel/commit/3205342a6a7209c55ca3a4bd97e986961fdc792e#diff-3048cb37b97e27515852c200994f3257b8ae33a465421d05184713377c0895b1R150)]
This PR also tightened the tolerance in case of encountering a similar issue.

## Testing Done

- Hardware Type: <BLANK>
- [X] run `make test` to ensure correctness
- [X] run `make checkstyle` to ensure code style
- [ ] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Review by shivam15s — CHANGES_REQUESTED** [review]

File: test/chunked_loss/test_cpo_loss.py [hunk 1]
```diff
@@ -59,13 +59,13 @@ def alignment_loss(
...
                 F.logsigmoid(self.beta * logits) * (1 - self.label_smoothing)
                 + F.logsigmoid(-self.beta * logits) * self.label_smoothing
             )
         elif self.loss_type == "simpo":
             logits = logits - (self.simpo_gamma / self.beta)
-            losses = (
+            losses = -(
                 F.logsigmoid(self.beta * logits) * (1 - self.label_smoothing)
                 + F.logsigmoid(-self.beta * logits) * self.label_smoothing
             )
```
[the remark this answers is not in the document]
Comment (austin362667) [reply test/chunked_loss/test_cpo_loss.py]: Sure! Thank you for reviewing.

**Review by yundai424 — APPROVED** [review]

**Review by shivam15s — APPROVED** [review]
